webapp/lib/controller.dart

```diff
@@ -372,22 +372,30 @@ void updateNeedsReplyCharts(List<model.NeedsReplyData> filteredNeedsReplyDataLis
   Map<DateTime, int> data = new Map.fromIterable(filteredNeedsReplyDataList,
     key: (item) => (item as model.NeedsReplyData).datetime.toLocal(),
     value: (item) => (item as model.NeedsReplyData).needsReplyCount);
-  view.contentView.needsReplyTimeseries.updateChart([data], timeScaleUnit: timeScaleUnit);
+  List<DateTime> timestamps = data.keys.toList()..sort();
+  DateTime xUpperLimitDateTime = new DateTime(timestamps.last.year, timestamps.last.month, timestamps.last.day + 1, 00);
```

Comment:
Consider turning this into a getter?

Reply:
Sorry, actually I'm not so sure of the logic here. I think the upper limit should be relative to `now()` and not to the latest timestamp?

Reply:
Yes, it should actually. I've seen the mistake. I hadn't thought about charts which may be out of date. I've corrected it to use `now()`